RecoMET/METFilters/plugins/EcalDeadCellTriggerPrimitiveFilter.cc

```diff
@@ -433,8 +440,13 @@ int EcalDeadCellTriggerPrimitiveFilter::setEvtRecHitstatus(const double &tpValCu
      if( !ebrechit->isRecovered() ) toDo = false;
 //     if( !ebrechit->checkFlag(EcalRecHit::kTowerRecovered) ) toDo = false;
 
+
+
      if( toDo ){
 
+	//If we considerkTPSaturated and a recHit has a flag set, we can immediately flag the event.
+        if(ebrechit->checkFlag(EcalRecHit::kTPSaturated) && usekTPSaturated_) return 1;
```

Comment:
@kencall would be more efficient to first check useTPSaturated_ here and below. Something to consider for your next PR.